server/src/main/java/org/opensearch/search/SearchService.java

```diff
@@ -927,6 +975,24 @@ public boolean freeReaderContext(ShardSearchContextId contextId) {
         return false;
     }
 
+    public void updatePitIdAndKeepAlive(
+        TransportCreatePITAction.UpdatePITContextRequest request,
+        ActionListener<TransportCreatePITAction.UpdatePitContextResponse> listener
+    ) {
+        try {
+            PitReaderContext readerContext = getPitReaderContext(request.getSearchContextId());
+            if (readerContext == null) {
+                throw new SearchContextMissingException(request.getSearchContextId());
+            }
+            readerContext.setPitId(request.getPitId());
+            readerContext.tryUpdateKeepAlive(request.getKeepAlive());
```

Comment:
what about `lastAccessTime` in ReaderContext? that should be updated to here I reckon.